docs/proposals/choosing-scheduler.md

```diff
@@ -0,0 +1,165 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest release of this document can be found
+[here](http://releases.k8s.io/release-1.1/docs/proposals/choosing-scheduler.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Choosing scheduler in a multi-scheduler configuration
+
+## Introduction
+
+This document proposes a mechanism for assigning a scheduler to a pod
+in a system which is using multiple schedulers as described [here](multiple-schedulers.md).
+In particular, we decribe an admission controller which adds the scheduler name annotation,
+and which takes its configuration from a new API object called `PodPolicy` that determines,
+for each pod, which scheduler is responsible for scheduling that pod, based on various
+characteristics of the pod.
+
+## Implementation
+
+We will create a new admission controller called PodSchedulerPolicy. It will be configured
+using a new API object called `PodPolicy`. We have chosen a generic name and API
+for this object because we want to use the same object for defining other policies based on pod
+characteristics, for example see #17097.
+
+PodSchedulerPolicy will iterate through the `Rules` of the `PodPolicy` and apply the
```

Comment:
I am not SURE it's better, but it sounded like it would solve more than one
such problem.  This is why I was confused about policy value - I was
looking for where you spelled out "this is a scheduler policy" vs "this is
a network policy".

I know Brian and Clayton have some plans around really complicated
LimitRange changes (com,plicated to my eyes anyway).  Maybe these should
start to fall in line more?

On Sun, Dec 6, 2015 at 11:40 PM, David Oppenheimer <notifications@github.com
> wrote:

> In docs/proposals/choosing-scheduler.md
> <https://github.com/kubernetes/kubernetes/pull/18262#discussion_r46792873>
> :
>
> > +
> > +This document proposes a mechanism for assigning a scheduler to a pod
> > +in a system which is using multiple schedulers as described [here](multiple-schedulers.md).
> > +In particular, we decribe an admission controller which adds the scheduler name annotation,
> > +and which takes its configuration from a new API object called `PodPolicy` that determines,
> > +for each pod, which scheduler is responsible for scheduling that pod, based on various
> > +characteristics of the pod.
> > +
> > +## Implementation
> > +
> > +We will create a new admission controller called PodSchedulerPolicy. It will be configured
> > +using a new API object called `PodPolicy`. We have chosen a generic name and API
> > +for this object because we want to use the same object for defining other policies based on pod
> > +characteristics, for example see #17097.
> > +
> > +PodSchedulerPolicy will iterate through the `Rules` of the `PodPolicy` and apply the
>
> I was very confused by "a new API object called PodPolicy" above.
>
> And rightly so. What I wrote is simply wrong -- very sorry about that.
> It's not an API object.
>
> . I assumed you were adding a generic PodPolicy API resource that carried
> a map of strings that various components could use for policy.
>
> Ah, that's an interesting idea. (And maybe that's what @bgrant0607
> <https://github.com/bgrant0607> had in mind when he suggested to do
> something eneric.) I can redo this proposal in that model.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/18262/files#r46792873>.
>